### PR TITLE
streamingccl: hookup tracing aggregtor events for the C2C job

### LINKED ISSUE #100126: jobsprofiler: periodically persist aggregated stats during job execution
Processors of some jobs have a `TracingAggregator` attached to their root ctx before starting execution. This aggregator is subscribed to listen for all StructuredEvents that are emitted in the associated trace. The aggregator maintains a rolling aggregate of the StructuredEvents it is notified about. Today, this rolling aggregate is held in memory and is thrown away once the job completes execution. With the introduction of the `job_info` table we should start periodically persisting these aggregated stats over the lifetime of the job. This will give us a timeseries of all the collected stats over the lifetime of the job. The information persisted in the job_info table can then be consumed at a future point by tooling that we build to analyze the performance of a job.

Epic: CRDB-8964

Jira issue: CRDB-26268

### PR DESCRIPTION
**Pull request by adityamaru**
This change builds on top of https://github.com/cockroachdb/cockroach/pull/107994 and wires up each stream
ingestion data processor to emit TracingAggregatorEvents to
the frontier and subsequently the job coordinator.
These events are periodically flushed to files in the `job_info`
table and are consumable via the DBConsole Job Details page.

Currently, the only aggregator event that is propagated is the
IngestionPerformanceStats emitted by the sst batcher.

Informs: #108374
Fixes: https://github.com/cockroachdb/cockroach/issues/100126
Release note: None

### CODE REVIEW

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go [hunk 1]
```diff
@@ -333,6 +340,34 @@ func newStreamIngestionDataProcessor(
...
+		}
+		data := make([]byte, msg.Size())
+		if _, err := msg.MarshalTo(data); err != nil {
+			// This should never happen but if it does skip the aggregated event.
+			log.Warningf(ctx, "failed to unmarshal aggregated event: %v", err.Error())
+			return
+		}
+		aggEvents.Events[name] = data
+	})
+	log.Infof(ctx, "constructing tracing aggregator %+v", aggEvents)
```
Comment (stevendanna) [inline pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go:366]: Is this log line useful to us?

File: pkg/sql/colexec/colbuilder/execplan.go [hunk 2]
```diff
@@ -313,7 +314,9 @@ func canWrap(mode sessiondatapb.VectorizeExecMode, core *execinfrapb.ProcessorCo
 	case core.RestoreData != nil:
 	case core.Filterer != nil:
 	case core.StreamIngestionData != nil:
+		return errStreamIngestionWrap
```
Comment (stevendanna) [inline pkg/sql/colexec/colbuilder/execplan.go:317]: Interesting. I read the linked issue and understand why we need this. I'm a little surprised we don't need this for the restore case though.

**Review by stevendanna — COMMENTED** [review]
This looks good, but it has produced another thought about some of the PRs it builds on. But once those PRs are in, I think this should be g2g.

File: pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go (hunk 1 above)
Comment (adityamaru) [reply pkg/ccl/streamingccl/streamingest/stream_ingestion_processor.go:366]: nope, deleted

File: pkg/sql/colexec/colbuilder/execplan.go (hunk 2 above)
Comment (adityamaru) [reply pkg/sql/colexec/colbuilder/execplan.go:318]: I am surprised too, let me dig some more.

**Review by stevendanna — APPROVED** [review]

File: pkg/sql/colexec/colbuilder/execplan.go (hunk 2 above)
Comment (adityamaru) [reply pkg/sql/colexec/colbuilder/execplan.go:318]: I think this has something to do with the fact that in c2c the stream ingestion procs push the meta downstream to another processor in the frontier proc. In restore however the restore data processor is the leaf processor and pushes metas to the row result writer. I'll add some logging to sanity check that the restore data procs aren't doing any unwanted buffering of progress metas.

**Comment by adityamaru** [comment]
bors r=stevendanna

**Pull request merged**